### PR TITLE
Deprecate `sky_diffuse_passias` and amend `plot_passias_diffuse_shading.py` gallery example

### LINKED ISSUE #2584: Diffuse Self-Shading Example: Incorrect transposition of shading.sky_diffuse_passias
**Describe the bug**
I think the second graph of https://pvlib-python.readthedocs.io/en/stable/gallery/shading/plot_passias_diffuse_shading.html is incorrect, even though it corresponds with the given reference.

In particular, transposing the shading loss in the horizontal plane `shading.sky_diffuse_passias(psi)` to the plane of array by multiplying it with a transposition ratio seems incorrect to me.

Instead the 'relative diffuse irradiance' (defined as 'ratio of diffuse plane of array irradiance (after accounting for shading) to diffuse horizontal irradiance') should be simply calculated as `shading.sky_diffuse_passias(surface_tilt + psi)`.

This result also corresponds (more or less) with the result obtained when using the view factor method `bifacial.utils.vf_row_sky_2d_integ(surface_tilt, gcr)`, which I guess is the most accurate method.

The difference between the various methods is illustrated with the following graph:

```python
from pvlib import bifacial, shading, irradiance
from cycler import cycler
import matplotlib.pyplot as plt
import numpy as np

surface_tilt = np.arange(0, 90, 0.5)

plt.rc('axes', prop_cycle=(cycler('color', ['b',  'orange', 'g', 'r']) * cycler('linestyle', ['-', ':', '--'])))
plt.figure()
for k in [1, 1.5, 2, 10]:
    gcr = 1/k
    psi = shading.masking_angle_passias(surface_tilt, gcr)
    shading_loss = shading.sky_diffuse_passias(psi)
    transposition_ratio = irradiance.isotropic(surface_tilt, dhi=1.0)
    passias = transposition_ratio * (1-shading_loss) * 100  # %
    plt.plot(surface_tilt, passias, label=f"passias k={k}")
    passias_corrected = 100 - 100 * shading.sky_diffuse_passias(surface_tilt + psi)
    plt.plot(surface_tilt, passias_corrected, label=f"passias corrected k={k}")
    vf_row_sky_2d_integ = bifacial.utils.vf_row_sky_2d_integ(surface_tilt, gcr) * 100 # %
    # hottel = (1 - shading.sky_diffuse_hottel(surface_tilt, gcr)) * 100 # %
    plt.plot(surface_tilt, vf_row_sky_2d_integ, label=f"vf_row_sky_2d_integ k={k}")

plt.xlabel('Inclination angle [degrees]')
plt.ylabel('Relative diffuse irradiance [%]')
plt.ylim(0, 105)
plt.legend()
plt.show()
```

Image

### PR DESCRIPTION
**Pull request by kdebrab**
- [x] Closes #2584
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing/index.html)
 - [ ] ~~Tests added~~
 - [ ] ~~Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [ ] ~~New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~~
 - [x] Pull request is nearly complete and ready for detailed review.
 - [ ] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: docs/examples/shading/plot_passias_diffuse_shading.py [hunk 1]
```diff
@@ -82,3 +89,37 @@
 # %%
 # As ``k`` decreases, GCR increases, so self-shading loss increases and
 # collected diffuse irradiance decreases.
+#
+# However, the pvlib-python authors believe that this approach is incorrect.
```
Comment (RDaxini) [inline docs/examples/shading/plot_passias_diffuse_shading.py:93]: What's the source of this and the other comments below? Was this discussed in an issue somewhere?

File: docs/examples/shading/plot_passias_diffuse_shading.py [hunk 2]
```diff
@@ -20,15 +20,21 @@
 # :py:func:`pvlib.shading.masking_angle_passias` and
 # :py:func:`pvlib.shading.sky_diffuse_passias`.
 #
+# However, the pvlib-python authors believe that this approach is incorrect.
```
Comment (RDaxini) [inline docs/examples/shading/plot_passias_diffuse_shading.py:23]: Same comment here as my other comment https://github.com/pvlib/pvlib-python/pull/2589/files#r2590549604

File: docs/examples/shading/plot_passias_diffuse_shading.py (hunk 1 above)
Comment (cwhanse) [reply docs/examples/shading/plot_passias_diffuse_shading.py:93]: https://github.com/pvlib/pvlib-python/discussions/1825
